File: core/src/main/java/com/larryhsiao/nyx/core/jots/PostedJot.java
package com.larryhsiao.nyx.core.jots;

import com.silverhetch.clotho.Source;

import java.sql.Connection;
import java.util.Calendar;
import java.util.Date;

/**
 * Source to build a Jot that just updated or created.
 */
public class PostedJot implements Source<Jot> {
    private final Source<Connection> db;
    private final Jot jot;

    public PostedJot(Source<Connection> db, Jot jot) {
        this.db = db;
        this.jot = jot;
    }

    @Override
    public Jot value() {
        if (jot.id() == -1) {
            final Calendar calendar = Calendar.getInstance();
            calendar.setTime(new Date(jot.createdTime()));
            return new NewJot(db, jot.content(), jot.location(), calendar, jot.mood()).value();
        } else {
            new UpdateJot(jot, db).fire();
            return jot;
        }
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/jots/PostedJot.kt
package com.larryhsiao.nyx.core.jots

import com.silverhetch.clotho.Source
import java.sql.Connection
import java.util.*

/**
 * Source to build a Jot that just updated or created.
 */
class PostedJot @JvmOverloads constructor(
    private val db: Source<Connection>,
    private val jot: Jot,
    private val updateVer: Boolean = true
) : Source<Jot?> {
    override fun value(): Jot {
        return if (jot.id() == -1L) {
            val calendar = Calendar.getInstance()
            calendar.time = Date(jot.createdTime())
            NewJot(
                db,
                jot.title(),
                jot.content(),
                jot.location(),
                calendar,
                jot.mood()
            ).value()
        } else {
            UpdateJot(jot, db, updateVer).fire()
            jot
        }
    }
}